src/App.tsx

```diff
@@ -1,15 +1,53 @@
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const [selectPerson, setSelectPerson] = useState<Person | null>(null);
+  const [highlightedIndex, setHighlightedIndex] = useState<number | null>(null);
+
+  const timer = useRef(0);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(people =>
+      people.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery]);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setSelectPerson(null);
+
+    window.clearTimeout(timer.current);
+
+    timer.current = window.setTimeout(() => {
+      setAppliedQuery(event.target.value);
+    }, 300);
```

Comment:
The debounce delay should be customizable via props, with a default value of 300ms. Consider adding a prop for the delay and using it here instead of the hardcoded 300ms value.